drivers/platform/aducm3029/uart.c

```diff
@@ -0,0 +1,399 @@
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define CLK_FREQ	26000000u
+#define BAUDS_NB	10u
+
+/** Stores calculated baud rates and parameters */
+const struct baud_desc baud_rates_26MHz[BAUDS_NB] = {
+	{9600, 1078, 3, 24, 3},
+	{19200, 1078, 3, 12, 3},
+	{38400, 1321, 3, 8, 2},
+	{57600, 1078, 3, 4, 3},
+	{115200, 1563, 1, 4, 3},
+	{230400, 1563, 1, 2, 3},
+	{460800, 1563, 1, 1, 3},
+	{921600, 1563, 1, 1, 2},
+	{1000000,1280, 1, 1, 2},
+	{1500000, 171, 1, 1, 2}
+};
+
+/**
+ * Buffer needed by the ADI UART driver to operate.
+ *  This buffer is aligned at runtime to 32 bits
+ */
+static uint8_t	*adi_uart_buffer = NULL;
+/** Stores the offset used to align adi_uart_buffer. Needed to deallocate it */
+static uint32_t	adi_uart_buffer_offset;
+/**
+ * Boolean value used in CALLBACK_MODE. Is set when \ref uart_read is called and
+ *  reset when the buffer is processed
+ */
+static uint32_t waiting_read_callback = 0;
+/**
+ * Boolean value used in CALLBACK_MODE. Is set when \ref uart_write is called
+ *  and reset when the buffer is processed
+ */
+static uint32_t waiting_write_callback = 0;
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Allocates the memory needed for the UART descriptor
+ * @return Address to the allocated memory, NULL if the allocation fails.
+ */
+static struct uart_desc *alloc_desc_mem(void)
+{
+	struct uart_desc	*desc;
+	struct aducm_uart_desc	*aducm_desc;
+
+	desc = calloc(1, sizeof(struct uart_desc));
+	if (!desc)
+		return NULL;
+	aducm_desc = calloc(1, sizeof(struct aducm_uart_desc));
+	if (!aducm_desc) {
+		free(desc);
+		return NULL;
+	}
+	desc->extra = aducm_desc;
+
+	/* Allocate and align buffer to 32 bits */
+	uint32_t mem;
+	mem = (uint32_t)calloc(1, ADI_UART_BIDIR_MEMORY_SIZE + 3);
+	if (!mem) {
+		free(aducm_desc);
+		free(desc);
+		return NULL;
+	}
+	adi_uart_buffer = (uint8_t *)((mem+3u) & (~(3u)));
+	adi_uart_buffer_offset = (uint32_t)adi_uart_buffer - mem;
+
+	return desc;
+}
+
+/**
+ * @brief Deallocates the memory needed for the UART descriptor
+ * @param desc - Descriptor to be deallocated
+ */
+static void free_desc_mem(struct uart_desc *desc)
+{
+	free((void *)((uint32_t)adi_uart_buffer - adi_uart_buffer_offset));
+	adi_uart_buffer = NULL;
+	adi_uart_buffer_offset = 0;
+	free(desc->extra);
+	free(desc);
+}
+
+/**
+ * @brief Call the user defined callback when a read/write operation completed
+ * @param pCBParam:	Client supplied callback parameter
+ * @param Event:	Event ID from ADI_UART_EVENT
+ * @param pArg:		Pointer to the handled buffer or to a error code
+ */
+static void uart_callback(void *pCBParam, uint32_t Event, void *pArg)
```

Comment:
Since this is C programming language, I would stick to the naming conventions:
- variable/function names with lowercase + underscores as separators.

http://manual.gromacs.org/documentation/5.1-current/dev-manual/naming.html#c-code

Reply:
done